src/api/java/com/minecolonies/api/tileentities/TileEntityStash.java

```diff
@@ -0,0 +1,74 @@
+package com.minecolonies.api.tileentities;
+
+import com.minecolonies.api.colony.buildings.IBuilding;
+import com.minecolonies.api.colony.buildings.IBuildingContainer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntityType;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nonnull;
+
+import static com.minecolonies.api.util.constant.BuildingConstants.MAX_PRIO;
+import static com.minecolonies.api.util.constant.Constants.DEFAULT_SIZE;
+
+/**
+ * Class which handles the tileEntity for the Stash block.
+ */
+
+public class TileEntityStash extends TileEntityColonyBuilding
+{
+
+    public TileEntityStash(final TileEntityType type)
+    {
+        super(type);
+        inventory = new NotifyingRackInventory(DEFAULT_SIZE);
+    }
+
+    public TileEntityStash()
+    {
+        super(MinecoloniesTileEntities.STASH);
+        inventory = new NotifyingRackInventory(DEFAULT_SIZE);
+    }
+
+    /**
+     * Called when the inventory of the tileEntity it holds it's changed
+     *
+     * @param isEmpty
+     */
+    private void buildingInventoryChanged(boolean isEmpty)
+    {
+        IBuildingContainer building = getBuilding();
+        if (!isEmpty && building != null && !building.isPriorityStatic())
+        {
+            if (building instanceof IBuilding)
```

Comment:
why do we need a different handling for IBuilding here? isn't it always an IBuilding or never?

Reply:
It should be always an instance of `IBuilding`, but as so far as I've looked, I haven't read the full project to see if it could be reset somehow, that's why I left the check there. Removing it.

Reply:
To be fair, I would replace the building != null check with that instanceof check.